### PR TITLE
feat(requests): Add tokens/s toggle to latency column

### PR DESCRIPTION
**Pull request by djdembeck**
## Overview | 概述

**English:**
Adds a toggle to the requests table latency column that allows users to switch between viewing latency (ms) and tokens per second (tok/s) for each request. The display preference is persisted in localStorage and includes SSR-safe initialization to prevent hydration mismatches.

image

**中文:**
在请求表的延迟列中添加了一个切换功能，允许用户在查看每个请求的延迟（毫秒）和每秒令牌数（tok/s）之间切换。显示偏好设置保存在 localStorage 中，并包含 SSR 安全初始化以防止水合不匹配。

---

### Frontend (`requests-columns.tsx`)
- Added interactive click handler to latency column header for toggling display mode
- Implemented tokens per second calculation with proper handling for streaming/non-streaming requests
- Used SwapHorizontal icon for the toggle button with accessibility support
- Added column sorting to maintain consistent user experience

### Utility Functions (`tokens-per-second.ts`)
- Created `calculateTokensPerSecond()` function with comprehensive edge case handling:
  - Handles null/undefined usage logs
  - Accounts for zero latency scenarios
  - Properly handles streaming vs non-streaming request latency calculation
  - Subtracts TTFT (time to first token) for streaming requests to get accurate generation throughput
- Implemented `useDisplayMode()` hook with:
  - SSR-safe two-phase initialization pattern
  - localStorage persistence with runtime validation
  - Type-safe display mode transitions

### Testing
- Added comprehensive unit tests (30+ test cases) covering:
  - Normal token calculation scenarios
  - Edge cases (null, undefined, empty arrays)
  - Streaming vs non-streaming behavior
  - TTFT edge cases
  - Rounding behavior verification

### Localization
- Added translation keys for both English and Chinese locales

---

### Tokens/s Calculation
The tokens per second calculation excludes Time To First Token (TTFT) for streaming requests:
```typescript
let effectiveLatencyMs = request.metricsLatencyMs;
if (request.stream && request.metricsFirstTokenLatencyMs != null) {
  if (request.metricsFirstTokenLatencyMs < request.metricsLatencyMs) {
    effectiveLatencyMs = request.metricsLatencyMs - request.metricsFirstTokenLatencyMs;
  }
}
```

### SSR Safety Pattern
```typescript
const [displayMode, setDisplayMode] = useState<DisplayMode>('latency');

useEffect(() => {
  if (typeof window !== 'undefined') {
    const stored = localStorage.getItem('requests-table-latency-display-mode');
    if (stored && VALID_DISPLAY_MODES.includes(stored as DisplayMode)) {
      setDisplayMode(stored as DisplayMode);
    }
  }
}, []);
```

---

## Use Cases | 使用场景

1. **Performance Analysis**: Quickly identify which requests generated tokens fastest
2. **Model Comparison**: Compare token generation speed across different models
3. **Channel Optimization**: Evaluate channel performance based on throughput

## Summary by CodeRabbit

* **New Features**
  * Toggle between latency and tokens-per-second in the requests table.
  * Localization added for tokens-per-second and toggle tooltips (English & Chinese).

* **Tests**
  * Comprehensive tests for tokens-per-second calculations and display-mode behavior.
  * Test setup and JS DOM environment configured.

* **Chores**
  * Updated React and React DOM and related type packages.
  * Added testing tooling and coverage configuration.

### CODE REVIEW

**Comment by djdembeck** [comment]
@coderabbitai review

File: frontend/package.json [hunk 1]
```diff
@@ -73,9 +73,7 @@
     "motion": "^12.23.24",
     "nanoid": "^5.1.5",
     "next-themes": "^0.4.6",
-    "react": "^19.2.0",
```
Comment (looplj) [inline frontend/package.json:76]: I think the react and react-dom should be placed in the dependencies, I am not sure if it is ok to place them in the devDependencies.

Comment (djdembeck) [reply frontend/package.json:76]: You are correct, and there were unintentional package upgrades I will roll back

**Comment by djdembeck** [comment]
Reverting the frontend testing, I think they would be best suited for a separate PR since they change dependencies.

**Review by looplj — APPROVED** [review]
